pkg/repo/fs.go

```diff
@@ -142,6 +143,12 @@ func (fsr *FsRepo) Open() error {
 		config.SetLibp2pKey(cfg.User.Libp2pKeyPath)
 	}
 
+	if cfg.User.InstallationID == "" {
+		ID, _ := config.GetClientID()
+		uuidFromUserID := uuid.NewSHA1(uuid.New(), []byte(ID))
+		config.SetUserID(uuidFromUserID.String())
+	}
+
```

Comment:
This is only the default that will be used when an installation ID is not explicitly set. The installation ID is meant to be set by the installer, so the common case is that it will be in config already.

Reply:
Ahh I see, is the code that explicitly sets the installationID going to follow this work? We'll need to pay close attention to how we persist it since setting the value in the config structure will not result in the file being written to disk/config file.